### DISCUSSION TITLE
NSPAWN interactive error: debian-custom failed with error code 1

### DISCUSSION THREAD

**Discussion by GigaBypass9876** [unifi-utilities/unifi-common#553] [Q&A]
UDM Pro SE running OS 3.0.18
In running the interactive guide to install nspawn, I selected Mac Vlan container, it ran for a long time installing packages, then stopped with this error:

"Container debian-custom exited successfully.
In container start network
Spawning container debian-custom on /data/custom/machines/debian-custom.
Press ^] three times within 1s to kill container.
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
Container debian-custom failed with error code 1"

I then went to the manual steps and realized it got through step 1 and began manually entering Step 2 as follows until the last command and got an error (below). I am reluctant to try the interactive script again since running all of those packages may produce duplicate files or otherwise make things worse. Can I do that? Is there a way to resume this interactive script or do I need to factory reset the UDM Pro SE and try it again. First, I am trying to understand why it failed in the first place. Please chime in if you think it will help. Thanks!

from @peacey: Step 2. Configure the Container
Now that the container is created, let's configure it. Make sure you are back on the host OS and not in the container.
1. First, we will link the container to /var/lib/machines so we can control it with machinectl.
mkdir -p /var/lib/machines
ln -s /data/custom/machines/debian-custom /var/lib/machines/
2. Next, we will create a debian-custom.nspawn file in /etc/systemd/nspawn to configure parameters for the container (such as network, bind mounts, etc). Here I use vim to create the file as a personal preference.
mkdir -p /etc/systemd/nspawn
vim /etc/systemd/nspawn/debian-custom.nspawn

I entered the commands above and got an error that "a swap file was already created" and not to make two instances of it.
"the editing session crashed"
I looked at the directory structure in Winscp, and directory seemingly created above,  /etc/systemd/nspawn, did not appear. Any way to resolve this without a system reset and trying again?

I am way out of my element on this one, so any help greatly appreciated.

**Comment by BOFHELL** [comment]
This script doesnt change the onboot 10-network... file and ure right the systemctl enable service failed ...

change line 119 ff

`echo "In container start network"
systemd-nspawn -M "$container_name" -D /data/custom/machines/"$container_name" /bin/bash -c systemctl enable systemd-networkd || true

echo "in container set default DNS 1.1.1.1"
systemd-nspawn -M "$container_name" --pipe -D /data/custom/machines/"$container_name" /bin/bash << EOF
echo "nameserver 1.1.1.1" > /etc/resolv.conf
echo "$container_name" > /etc/hostname
EOF


#### we will link the container to /var/lib/machines so we can control it with machinectl`

after this u have to change the /data/onboot.d/10-network-setup.sh with your variables <- after running the script
